news/2020-09-15-CDI_Lite_extension.adoc

```diff
@@ -0,0 +1,287 @@
+---
+title: "Build Compatible Extensions Proposal"
+summary: Description of an extension SPI for CDI lite
+layout: post
+authors: [ladt]
+tags: [Proposal]
+desc: Build Compatible Extensions Proposal
+priority: 1
+change_frequency: daily
+---
+
+
+A few months back we shared our vision for CDI lite.
+To make it short the goal with CDI lite is to make the spec lighter, cloud-friendlier and allow for build-time implementations which are now on the rise.
+This will be done by identifying a subset of features that can be implemented in build time environment.
+
+The idea is to keep the CDI programming model everyone loves but allow implementations to produce lighter and faster binaries.
+
+One issue we raised in the last post is that this build-time approach is not compatible with contemporary CDI portable extension model. Therefore, with existing CDI features we can only imagine CDI Lite without extension mechanism. Knowing the great popularity of portable extensions and their importance for integrators, we decided to revisit what could be done in that direction.
+
+We have worked to create a new extension SPI that could be used in Lite and today we are sharing this proposal with you. It is still a work in progress but should be a good enough starting point to design an extension mechanism for CDI lite.
+
+When we started thinking about a lite API design space, we mostly constrained ourselves to one particularly common use case: annotation transformations. Many things that Portable Extensions allow can be achieved simply by adding an annotation here or removing an annotation there. We know that other transformations are necessary, but the API for annotation transformation is the most developed proposal we have. Other bits and pieces are in various stages of construction, especially registering synthetic components. So, please, bear with us and submit feedback! (*TODO* where?)
```

Comment:
The paragraphs above this one, and the beginning of this one, are so different from my style of writing :-) Few comments:

- Is it "lite" or "Lite"? We should be coinsistent.
- Do we call it "Build Compatible Extensions" or not? I do in the rest of the text.
- Really not sure what's wrong with "For our explorations of the API design space, we mostly constrained ourselves ..." :-)